wire-ios-data-model/Source/MLS/OneOnOne/OneOnOneMigrator.swift

```diff
@@ -48,31 +48,35 @@ public struct OneOnOneMigrator: OneOnOneMigratorInterface {
         userID: QualifiedID,
         in context: NSManagedObjectContext
     ) async throws -> MLSGroupID {
+        // Fetch MLS 1:1 conversation and store it locally.
         let (mlsGroupID, removalKeys) = try await syncMLSConversationFromBackend(
             userID: userID,
             in: context
         )
 
-        if try await mlsService.conversationExists(groupID: mlsGroupID) {
-            return mlsGroupID
-        }
-
-        guard let epoch = await fetchMLSConversationEpoch(mlsGroupID: mlsGroupID, in: context) else {
-            throw MigrateMLSOneOnOneConversationError.missingConversationEpoch
-        }
+        // Create or join the MLS conversation if needed.
+        if try await !mlsService.conversationExists(groupID: mlsGroupID) {
+            guard let epoch = await fetchMLSConversationEpoch(mlsGroupID: mlsGroupID, in: context) else {
+                throw MigrateMLSOneOnOneConversationError.missingConversationEpoch
+            }
 
-        if epoch == 0 {
-            try await establishMLSGroupIfNeeded(
-                userID: userID,
-                mlsGroupID: mlsGroupID,
-                removalKeys: removalKeys,
-                in: context
-            )
-        } else {
-            try await mlsService.joinGroup(with: mlsGroupID)
+            if epoch == 0 {
+                try await establishMLSGroupIfNeeded(
+                    userID: userID,
+                    mlsGroupID: mlsGroupID,
+                    removalKeys: removalKeys,
+                    in: context
+                )
+            } else {
+                try await mlsService.joinGroup(with: mlsGroupID)
+            }
         }
```

Comment:
move to own private function
func createOrJoinMLSConversationIfNeeded.

Reply:
Done